scripts/wallet.js

```diff
@@ -74,9 +75,41 @@ export class Wallet {
      * @type {Boolean}
      */
     #isMainWallet;
+    /**
+     * Set of unique representations of Outpoints that keep track of locked utxos.
+     * @type {Set<String>}
+     */
+    #lockedCoins;
     constructor(nAccount, isMainWallet) {
         this.#nAccount = nAccount;
         this.#isMainWallet = isMainWallet;
+        this.#lockedCoins = new Set();
+    }
+
+    /**
+     * Check whether a given outpoint is locked
+     * @param {COutpoint} opt
+     * @return {Boolean} true if opt is locked, false otherwise
+     */
+    isCoinLocked(opt) {
+        return this.#lockedCoins.has(opt.toUnique());
+    }
+
+    /**
+     * Lock a given Outpoint
+     * @param {COutpoint} opt
+     */
+    lockCoin(opt) {
```

Comment:
I don't like this circular dependency, `Mempool` depends on `Wallet` for locked coins, and `Wallet` depends on `Mempool` for balance, in my opinion:
- `Wallet` should have a private `#mempool` instance
- `Mempool` shouldn't know about locked coins
- Have `Wallet::getBalance` calculate the balance.

But I'm fine to have this implemented in a subsequent PR since it's quite a big refactor